RHOSP-18.0.3/cinder/powerflex/README.md

```diff
@@ -0,0 +1,295 @@
+
+```
+vi storage_csm_powerflex_v292.yaml
+... [Truncated]
+    initContainers:
+      - image: dellemc/sdc:4.5
+        imagePullPolicy: IfNotPresent
+        name: sdc
+        envs:
+          - name: MDM
+            value: "MDM_IP"
+... [Truncated]
+```
+Apply the CR file to the OpenShift cluster. 
+```
+oc create -f storage_csm_powerflex_v2110.yaml
+```
+
+Confirm the SDC are now running as containers on worker nodes
+```
+[admin@rhadmin ~]$ oc get pods -n vxflexos
+NAME                                  READY   STATUS    RESTARTS   AGE
+vxflexos-controller-cb9855d69-7p9d4   5/5     Running   0          31d
+vxflexos-node-9f69t                   2/2     Running   2          31d
+vxflexos-node-kgrkd                   2/2     Running   2          31d
+vxflexos-node-vzfpm                   2/2     Running   2          31d
+[admin@rhadmin ~]$
+```
+
+To [install SDC](https://www.dell.com/support/manuals/en-ie/scaleio/powerflex_install_upgrade_guide_4.5.x/install-the-storage-data-client-on-a-linux-based-server?guid=guid-edaac602-f18b-4fe6-b825-ee09c6cdddd1&lang=en-us) on the compute(EDPM) nodes
+
+* Get the MDM IP's from PowerFlex
+* Copy the EMC-ScaleIO-sdc-*.rpm which corresponds to your RHEL OS level version to the EDPM nodes
+* Install the RPM as the root user
+* Repeat the steps for every remaining EDPM node
+* Confirm the SDC is connected to the PowerFlex system by logging on to PowerFlex Manager, navigate to Block → Hosts 
+
+## Configure the connector
+
+Before using attach/detach volume operations PowerFlex connector must be properly configured. Before the daaplane is installed, on each of the EDPM nodes do the following:
+
+Create `/opt/emc/scaleio/openstack/connector.conf` if it does not exist.
+
+```bash
+$ touch /opt/emc/scaleio/openstack/connector.conf
+```
+Edit the `/opt/emc/scaleio/openstack/connector.conf` and populate it with passwords.
+
+Example:
+
+```
+[powerflex]
+san_password = powerflex_password
+
+[powerflex-new]
+san_password = powerflex_password
+```
+
+Before the dataplane is deployed, in the `dataplane-nodeset.yaml` enter the following
+```
+edpm_network_config_template: |
+...
+edpm_nova_extra_bind_mounts:
+  - src: /opt/emc/scaleio/openstack/connector.conf
+    dest: /opt/emc/scaleio/openstack/connector.conf
+    options: "ro"
+```
+
+Create PowerFlex connector secret 
+```yaml
+vi powerflex-connector-secret.yaml
+
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    component: cinder-volume
+    service: cinder
+  name: powerflex-connector-conf-files
+stringData:
+  connector.conf: |
+    [powerflex]
+    san_password=PF_Manager_password
+type: Opaque
+```
+
+Edit the existing OpenStackControlPlane CR and add an extraMount specification as follows
+```
+...  
+spec:
+  extraMounts:
+  - extraVol:
+    - extraVolType: powerflex
+      mounts:
+      - mountPath: /opt/emc/scaleio/openstack
+        name: powerflex-connector
+        readOnly: true
+      propagation:
+      - CinderVolume
```

Comment:
Does it need to be propagated ton CinderBackup if Backup is enabled?

Reply:
We are using NFS for cinderback hence we didnt do any propagation for CinderBackup